8284047: Harmonize/Standardize the SSLSocket/SSLEngine/SSLSocketSSLEngine test templates

mpdonova opened the pull request:
* Refactored SSLContextTemplate and SSLSocketTemplate to put common code in one base class (SSLContextTemplate)
* Updated TLS/SSL tests to extend SSLSocketTemplate where possible.
* Updated SSLEngineTemplate to accommodate changes in SSLContextTemplate. To keep this changeset to a reasonable size, updates to SSLEngine tests will be made under JDK-8301194.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8284047](https://bugs.openjdk.org/browse/JDK-8284047): Harmonize/Standardize the SSLSocket/SSLEngine/SSLSocketSSLEngine test templates


### Reviewers
 * [Rajan Halade](https://openjdk.org/census#rhalade) (@rhalade - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12555/head:pull/12555` \
`$ git checkout pull/12555`

Update a local copy of the PR: \
`$ git checkout pull/12555` \
`$ git pull https://git.openjdk.org/jdk pull/12555/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12555`

View PR using the GUI difftool: \
`$ git pr show -t 12555`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12555.diff">https://git.openjdk.org/jdk/pull/12555.diff</a>

</details>

rhalade commented on test/jdk/javax/net/ssl/templates/SSLSocketSSLEngineTemplate.java:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2011, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2011, 2023, Oracle and/or its affiliates. All rights reserved.
```
Can you please check test `sun/security/ssl/SSLEngineImpl/SSLEngineBadBufferArrayAccess.java` as this template test was a copy of it? Suggest you to check if this template copy can be removed now as we have other template classes for SSLSocket/SSLEngine client and server.

mpdonova replied:
Removed SSLSocketSSLEngine.java

rhalade commented on test/jdk/javax/net/ssl/ALPN/SSLServerSocketAlpnTest.java:
```diff
@@ -28,7 +28,7 @@
  * @test
  * @bug 8051498 8145849 8158978 8170282
  * @summary JEP 244: TLS Application-Layer Protocol Negotiation Extension
- * @compile MyX509ExtendedKeyManager.java
```
Suggest you to delete file `MyX509ExtendedKeyManager.java` if it is not needed anymore.

mpdonova replied:
removed

rhalade replied:
Don't see this file deleted in PR.

mpdonova replied:
It was still being used by an "engine" test in that directory; I removed that last use and deleted the key manager class.

(there is another bug to refactor the SSLEngine tests.)

rhalade commented on test/jdk/javax/net/ssl/TLSCommon/TLSWithEdDSA.java:
```diff
@@ -263,7 +263,7 @@ private static void keyCheck(PublicKey pubKey, String expPkType,
      */
     @Override
-    protected SSLContext createClientSSLContext() throws Exception {
+    public SSLContext createClientSSLContext() throws Exception {
```
Why is this change needed, may be unintentional?

mpdonova replied:
reverted the file

rhalade commented on test/jdk/javax/net/ssl/templates/SSLContextTemplate.java:
```diff
@@ -39,534 +39,892 @@
+
+        CA_RSA_512( // for DisabledShortRSAKeys test
+                "RSA",
+                "-----BEGIN CERTIFICATE-----\n" +
```
Please add a comment with validity and algorithms similar to other entries in this enum. There are few other entries to update as well.

rhalade commented on test/jdk/sun/net/www/protocol/https/HttpsClient/ServerIdentityTest.java:
```diff
@@ -108,12 +108,12 @@ protected void runClientApplication(int serverPort) throws Exception {
 
     @Override
-    protected SSLContext createServerSSLContext() throws Exception {
+    public SSLContext createServerSSLContext() throws Exception {
```
revert his as well.

mpdonova replied:
reverted

rhalade commented on test/jdk/sun/security/ssl/SignatureScheme/SigAlgosExtTestWithTLS13.java:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2021 THL A29 Limited, a Tencent company. All rights reserved.
+ * Copyright (C) 2021, 2023 THL A29 Limited, a Tencent company. All rights reserved.
```
Missing comma.

rhalade replied:
May be you can revert this file, change is unintentional.

mpdonova replied:
reverted

rhalade commented on test/jdk/javax/net/ssl/ALPN/SSLServerSocketAlpnTest.java:
```diff
@@ -178,13 +133,15 @@ void doServerSide() throws Exception {
+     * If the server prematurely exits, serverReady will be set to true
+     * to avoid infinite hangs.
+     */
+    protected void runServerApplication(SSLSocket sslSocket) throws Exception {
```
Missing `@Override`

rhalade commented on test/jdk/javax/net/ssl/ALPN/SSLServerSocketAlpnTest.java:
```diff
@@ -278,7 +218,16 @@ void doClientSide() throws Exception {
+     * If the server prematurely exits, serverReady will be set to true
+     * to avoid infinite hangs.
+     */
+    protected void runClientApplication(SSLSocket sslSocket) throws Exception {
```
Missing `@Override`. Comment applies to few other files in this review.

rhalade commented on test/jdk/javax/net/ssl/ALPN/SSLServerSocketAlpnTest.java:
```diff
@@ -84,51 +84,14 @@
 
```
`java.security.KeyStore` is unused

rhalade commented on test/jdk/javax/net/ssl/TLSv12/DisabledShortRSAKeys.java:
```diff
@@ -58,70 +58,29 @@
 
```
With the update to this file, many imports are unused.

rhalade commented on test/jdk/sun/security/ssl/CipherSuite/NamedGroupsWithCipherSuite.java:
```diff
@@ -119,12 +119,12 @@ public class NamedGroupsWithCipherSuite extends SSLSocketTemplate {
     }
 
-    protected SSLContext createClientSSLContext() throws Exception {
+    public SSLContext createClientSSLContext() throws Exception {
```
please revert unnecessary change to this file

rhalade commented on test/jdk/sun/security/ssl/SSLContextImpl/MultipleChooseAlias.java:
```diff
@@ -68,7 +68,7 @@ protected void configureServerSocket(SSLServerSocket socket) {
 
     @Override
-    protected ContextParameters getClientContextParameters() {
+    public ContextParameters getClientContextParameters() {
```
Please revert unnecessary change to this file

rhalade commented on test/jdk/sun/security/ssl/SSLContextImpl/TrustTrustedCert.java:
```diff
@@ -113,7 +113,7 @@ public class TrustTrustedCert extends SSLSocketTemplate {
 
     @Override
-    protected SSLContext createServerSSLContext() throws Exception {
+    public SSLContext createServerSSLContext() throws Exception {
```
Same here, no change is needed in this file.

rhalade reviewed (changes requested):
Updated review looks good to me except few minor code cleanup comments.

mpdonova commented:
> Updated review looks good to me except few minor code cleanup comments.

I went through all the changed files and added @Override where necessary and cleaned up imports. I think I got them all.

mpdonova commented:
/integrate

rhalade commented:
/sponsor
